### PR TITLE
Switch to ChooseStreamAuto setting and add context menu

### PR DESCRIPTION
**Pull request by i96751414**
This will allow you to choose stream automatically when ChooseStreamAuto is disabled (using context menu)

### CODE REVIEW

**Comment by i96751414** [comment]
Btw, I couldn't test this one because I've got no working providers. But it was working on my last pulsar version

**Comment by scakemyer** [comment]
Ok I really don't want to give you a hard time or anything, just need to understand this properly. Took me a few minutes to understand the switcharoo that this is doing and still not sure I got it right.

Just for the record, I do prefer how you're calling this option "Choose Stream automatically", and defaulting it to `true` to match the previous behaviour. Now if I understand correctly, if I choose to disable automatic Choose Stream, Play now opens the Choose Stream dialog, and it also switches the Choose Stream context menu to automatically play? I think I understand why you're doing this, so that the option to use the automatic chooser is still readily available, but this gets pretty confusing considering the same context menu now has two different behaviours. One setting should change the behaviour of only one thing, in this case "Play". I like the idea of allowing the user to make it choose automatically also, but it would have to be labelled as such, and as a separate context menu option, keeping the current "Choose Stream" to show the same dialog, regardless of if Play also does.

I hope you understand what I mean to say here, because I like where you're going with this, it just feels too confusing the way it switches the behaviour of both menu items.

**Comment by i96751414** [comment]
You're right. Yes I have also thought about changing the label but then i have remembered that pulsar saves everything in cache, so changing the labels would be a bad idea...

**Comment by scakemyer** [comment]
Agreed, I was thinking more like showing another menu item when Choose Stream Automatically is disabled, I think using that same string would be fine here. You enable or disable it globally, which changes the behaviour of Play, but both "Choose Stream" and "Choose Stream Automatically" could be available in the context menu. I think it's ok if they're just always there also, no need to hide the latter when the global setting is disabled.

**Comment by i96751414** [comment]
It's ok for me... I was putting it like that because we should not 'complicate' things, I mean, we should use context menu options as less as possible. (Also, I will add another context menu option soon, because of Library Integration)

Should I close this and send a new PR?

**Comment by scakemyer** [comment]
Just push new commits or rebase/squash them, no need to make a new PR for pretty much the same feature.

File: api/movies.go [hunk 1]
```diff
@@ -93,7 +94,7 @@ func renderMovies(movies tmdb.Movies, ctx *gin.Context, page int) {
 		items = append(items, item)
 	}
 	if page >= 0 {
-		path := ctx.Request.URL.Path
+		path := ctx.Request.URL.Path
```
Comment (scakemyer) [inline api/movies.go]: Before your next commit, little nitpicking about the extra whitespace ;)

**Comment by scakemyer** [comment]
For reference, scakemyer/plugin.video.quasar#15

**Comment by scakemyer** [comment]
This looks great. Don't mind the failed builds for now, working on that.

**Pull request merged**